pandas/tests/frame/test_alter_axes.py

```diff
@@ -234,9 +234,16 @@ def test_set_index_pass_arrays_duplicate(
 
         # need to adapt first drop for case that both keys are 'A' --
         # cannot drop the same column twice;
-        # use "is" because == would give ambiguous Boolean error for containers
+        # plain == would give ambiguous Boolean error for containers
         first_drop = (
-            False if (keys[0] is "A" and keys[1] is "A") else drop  # noqa: F632
+            False
+            if (
+                type(keys[0]) == str
```

Comment:
use isinstance(keys[0], str) (and same below)

Reply:
That's further from what the 'is' did, but given the values it can take ("A" is the only string-like, the others are ['foo','foo','foo','bar','bar'] as various types) it probably makes no difference, so done (but not tested until the CI gets to it).